pilot/test/xdstest/extract.go

```diff
@@ -68,6 +69,49 @@ func ExtractRoutesFromListeners(ll []*listener.Listener) []string {
 	return routes
 }
 
+func ExtractClusterSecretResources(t test.Failer, c *cluster.Cluster) []string {
+	resourceNames := sets.New[string]()
+	var sockets []*core.TransportSocket
+	if c.TransportSocket != nil {
+		sockets = append(sockets, c.TransportSocket)
+	}
+	for _, ts := range c.TransportSocketMatches {
+		sockets = append(sockets, ts.TransportSocket)
```

Comment:
nit: for proxy protocol, this is not very correct. But since only used test. Doesnot matter much

Reply:
Haven't investigated this yet, can we update it when we need?